pkg/cloud/powervs.go

```diff
@@ -235,7 +236,7 @@ func (p *powerVSCloud) GetImageByID(imageID string) (*PVMImage, error) {
 
 func (p *powerVSCloud) CreateDisk(volumeName string, diskOptions *DiskOptions) (disk *Disk, err error) {
 	var volumeType string
-	capacityGiB := util.BytesToGiB(diskOptions.CapacityBytes)
+	capacityGiB := float64(util.BytesToGiB(diskOptions.CapacityBytes))
```

Comment:
this looks to be a little odd, we are converting int64 => float64 and later in the end again back to int64, instead we can just change to float64 directly while assigning to the field, e.g:

```
	dataVolume := &models.CreateDataVolume{
 		Name:      &volumeName,
 		Size:      pointer.Float64Ptr(float64(capacityGiB)),
 		Shareable: &diskOptions.Shareable,
 		DiskType:  volumeType,
 	}
```
> Note: pointer module is from https://github.com/kubernetes/utils/blob/master/pointer/pointer.go

or change the `CapacityGiB` type to float64 if that is more convenient.

Reply:
fixed